core/src/test/java/org/apache/calcite/rex/RexProgramTest.java

```diff
@@ -1687,10 +1687,8 @@ private void checkExponentialCnf(int n) {
         isNotNull(bRef));
     // [CALCITE-4352] causes "and b is not null" to disappear from the expanded
     // form.
-    final String simplified = "AND(SEARCH($0, Sarg[(0..10)]),"
-        + " SEARCH($1, Sarg[NOT NULL]))";
-    final String expanded =
-        "AND(AND(>($0, 0), <($0, 10)), IS NOT NULL($1))";
+    final String simplified = "AND(SEARCH($0, Sarg[(0..10)]), IS NOT NULL($1))";
+    final String expanded = "AND(AND(>($0, 0), <($0, 10)), IS NOT NULL($1))";
     checkSimplify(expr, simplified)
```

Comment:
Do you think we should flatten those ANDs in `AND(AND(>($0, 0), <($0, 10)), IS NOT NULL($1))`? I know it might be difficult to achieve efficiently.

Reply:
I agree, have promoted it in the new commit.